README.md

```diff
@@ -1,81 +1,104 @@
+- [Install Terraform 1.0+](https://learn.hashicorp.com/tutorials/terraform/install-cli#install-terraform), e.g.:
+  + Brew (Homebrew/Mac OS): `brew tap hashicorp/tap && brew install hashicorp/tap/terraform`
+  + Choco (Chocolatey/Windows): `choco install terraform`
+  + Conda (Anaconda): `conda install -c conda-forge terraform`
+  + Debian (Ubuntu/Linux):
+    ```
+    sudo apt-get update && sudo apt-get install -y gnupg software-properties-common curl
+    curl -fsSL https://apt.releases.hashicorp.com/gpg | sudo apt-key add -
+    sudo apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
+    sudo apt-get update && sudo apt-get install terraform
+    ```
+- Create an account with any supported cloud vendor and expose its [authentication credentials via environment variables](https://registry.terraform.io/providers/iterative/iterative/latest/docs/guides/authentication)
 
-Have a feature request or found a bug? Let us know via [GitHub issues](https://github.com/iterative/terraform-provider-iterative/issues). Have questions? Join our [community on Discord](https://discord.gg/bzA6uY7); we'll be happy to help you get started!
+### Define a Task
 
-## License
+In a project root directory, create a file named `main.tf` with the following contents:
 
-Iterative Provider is released under the [Apache 2.0 License](https://github.com/iterative/terraform-provider-iterative/blob/master/LICENSE).
-
-## Development
-
-### Install Go 1.17+
+```hcl
+terraform {
+  required_providers { iterative = { source = "iterative/iterative" } }
+}
+provider "iterative" {}
+resource "iterative_task" "example" {
+  cloud   = "aws" # or any of: gcp, az, k8s
+  machine = "m"   # medium, or any of: l, xl, m+k80, xl+v100, ...
+
+  storage {
+    workdir = "."
+    output  = "results"
+  }
+  script = <<-END
+    #!/bin/bash
+    mkdir results
+    echo "Hello World!" > results/greeting.txt
+  END
+}
+```
 
-Refer to the [official documentation](https://golang.org/doc/install) for specific instructions.
+See [the reference](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#argument-reference) for the full list of options for `main.tf` -- including more information on [`machine` types](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#machine-type) with and without GPUs.
 
-### Clone the repository
+Run this once (in the directory containing `main.tf`) to download the `required_providers`:
 
 ```console
-git clone https://github.com/iterative/terraform-provider-iterative
-cd terraform-provider-iterative
+terraform init
 ```
 
-### Install the provider
+### Run a Task
 
-Build the provider and install the resulting binary to the [local mirror directory](https://www.terraform.io/docs/cli/config/config-file.html#implied-local-mirror-directories):
-
-```console
-make install
-```
+- `terraform apply`: launch cloud instance(s), upload `workdir`, and run `script`
```

Comment:
It might be confusing a bit. Should user upload `workdir` after `terraform apply`? This how I read this for the 1st time.

I'd suggest keeping one line per action/command as we have above. Like:

### Launch an training job

```
terraform apply
```

This command creates an instance in a cloud based on `main.tf` specification. After the instance is running it uploads the current directory (`workdir`) and data directory (we don't have this one, right? :) ) to the instance and executes your training script (`script`).

In case of spot instances, the cloud will run recovery logic using the auto-scaling group. 

Once the script is finishes (or fails) the instance will be terminated automatically. Results and logs are periodicaly syncing to cloud storage.

Like this....

Reply:
It should work e2e and show the use case. Otherwise, we are increasing time-to-value quite a lot. One of the most important metrics.

Reply:
better now?